packages/perseus-core/src/parse-perseus-json/general-purpose-parsers/defaulted.ts

```diff
@@ -2,10 +2,10 @@ import {success} from "../result";
 
 import type {Parser} from "../parser-types";
 
-export function defaulted<T, Default>(
+export function defaulted<T>(
     parser: Parser<T>,
-    fallback: (missingValue: null | undefined) => Default,
-): Parser<T | Default> {
+    fallback: (missingValue: null | undefined) => NoInfer<T>,
```

Comment:
TIL about `NoInfer<T>` (I read https://www.totaltypescript.com/noinfer).  

So this blocks the fallback's return value from being included in the type inference for `T`, right? So the return value's `T` will be only inferred from the `parser` type.

Reply:
Correct! FWIW writing this code made me happy because I think it's a _perfect_ place to use `NoInfer`.